tests/test_estimator.py

```diff
@@ -74,29 +93,128 @@ def test_free_energy_estimator():
         seed
     )
 
+    beta = 0.125
+
     lambda_schedule = np.linspace(0, 1.0, 4)
 
-    for client in [None, CUDAPoolClient(1)]:
+    def loss_fn(sys_params):
 
-        mdl = estimator.FreeEnergyModel(
+        endpoint_correct = False
+        mdl = estimator_abfe.FreeEnergyModel(
             unbound_potentials,
+            endpoint_correct,
             client,
             box,
             x0,
             v0,
             integrator,
+            barostat,
             lambda_schedule,
             100,
             100,
-            barostat
+            beta,
+            "test"
         )
 
-        value_and_grad_fn = jax.value_and_grad(estimator.deltaG, argnums=1, has_aux=True)
-        (dG, _), sys_grad = value_and_grad_fn(mdl, sys_params) # run fwd, store result, and run bwd
+        dG, bar_dG_err, results = estimator_abfe.deltaG(mdl, sys_params)
 
-        grad_fn = jax.grad(estimator.deltaG, argnums=1, has_aux=True)
-        grad, _ = grad_fn(mdl, sys_params)
+        return dG**2
+
+
+    for client in [None, CUDAPoolClient(1)]:
+        
+        value_and_grad_fn = jax.value_and_grad(loss_fn)
+        dG, sys_grad = value_and_grad_fn(sys_params) # run fwd, store result, and run bwd
+
+        grad_fn = jax.grad(loss_fn)
+        grad = grad_fn(sys_params)
 
         assert len(grad) == 2
         assert grad[0].shape == sys_params[0].shape
+        assert grad[1].shape == sys_params[1].shape
+
+
+def test_free_energy_estimator_with_endpoint_correction():
+
```

Comment:
```suggestion
def test_free_energy_estimator_with_endpoint_correction():
    """assert shape compatibility between gradients and system parameters"""
```

Reply:
Will expand on this a little in the code.

Reply:
addressed in `9336d6c`